pyperplan-upf/solver.py

```diff
@@ -13,33 +13,167 @@
 # limitations under the License.
 
 
+import search
 import os
 import subprocess
 import re
+from typing import List
 import upf
 from upf.problem_kind import ProblemKind
+from upf.problem import Problem
 from upf.io.pddl_writer import PDDLWriter
-from upf.exception import UPFException
+from upf.exceptions import UPFException
+from upf.action import Action, ActionParameter
+from upf.fnode import FNode
+from upf.types import Type as UpfType
+from upf.object import Object as UpfObject
+from upf.plan import ActionInstance, SequentialPlan
+
+from pddl.parser import DomainDef, ProblemDef, InitStmt, GoalStmt, ActionStmt, Variable
+from pddl.parser import Formula, Keyword, RequirementsStmt, Predicate, PredicatesStmt, PredicateInstance
+from pddl.parser import PreconditionStmt, EffectStmt
+from pddl.parser import Type as PyperplanType
+from pddl.parser import Object as PyperplanObject
+from pddl.tree_visitor import TraversePDDLProblem, TraversePDDLDomain
+
+from pyperplan import _ground, _search, _write_solution, SEARCHES, HEURISTICS
+
+from upf_pyperplan.converter import ExpressionConverter
 
 
 class SolverImpl(upf.Solver):
     def __init__(self, weight=None, heuristic=None, **options):
-        pass
+        self._converter = ExpressionConverter()
+
+    # def solve(self, problem):
+    #     pddl_writer = PDDLWriter(problem)
+    #     pddl_writer.write_domain("pyperplan_domain.pddl")
+    #     pddl_writer.write_problem("pyperplan_problem.pddl")
+    #     cmd = "python3 pyperplan.py pyperplan_domain.pddl pyperplan_problem.pddl"
+    #     res = subprocess.run(cmd, capture_output=True)
+
+    #     if not os.path.isfile("pyperplan_problem.pddl.soln"):
+    #         print(res.stderr.decode())
+    #     else:
+    #         plan = self._plan_from_file(problem, "pyperplan_problem.pddl.soln")
+
+    #     return plan
+
+    def solve(self, problem: Problem):
+        domAST = self.parse_domain(problem)
```

Comment:
Why not creating a `pddl.pddl.Problem` object directly?
The only hard part seems splitting effects in positive and negative..